vignettes/articles/multi-scale-resampling.Rmd

```diff
@@ -0,0 +1,233 @@
+
+Because we're mostly interested in assessing our models, I'm going to mostly ignore how exactly I downloaded and wrangled the FIA data for this post. If you're curious, I've hidden the code below:
+
+<details>
+<summary>Pre-processing code</summary>
+
+```{r, eval=FALSE}
+library(dplyr)
+
+# Download the FIA database for New York over the internet,
+# and unzip it into our local directory
+#
+# This updates annually, which means that this post likely won't
+# generate the exact same results after 2022
+httr::GET(
+  "https://apps.fs.usda.gov/fia/datamart/Databases/SQLite_FIADB_NY.zip",
+  httr::write_disk("SQLite_FIADB_NY.zip", TRUE)
+)
+
+unzip("SQLite_FIADB_NY.zip")
+
+# We're going to work with the database through dplyr's database connections
+#
+# But first, we need to create a DBI connection to the database and
+# load out tables:
+con <- DBI::dbConnect(RSQLite::SQLite(), dbname = "FIADB_NY.db")
+trees <- tbl(con, "TREE")
+
+plots <- tbl(con, "PLOT")
+
+# The FIA database has every measurement ever collected by the program;
+# we'll filter to only the most recent survey for each of the plots.
+#
+# Plots are measured on a rolling 7 year basis, so we'll also cut out any
+# plots which might not be remeasured anymore with a call to filter()
+plots <- plots |> 
+  group_by(PLOT) |> 
+  filter(INVYR == max(INVYR, na.rm = TRUE)) |> 
+  ungroup() |> 
+  filter(INVYR > 2009) |> 
+  collect()
+
+copy_to(con, plots, "newest_plots", TRUE)
+newest_plots <- tbl(con, "newest_plots")
+
+# Now we'll use a filtering join to select only trees measured in the most
+# recent sample at each plot
+#
+# We'll also count how many trees were at each plot,
+# sum up their AGB, 
+# and save out a few other useful columns like latitude and longitude
+plot_measurements <- trees |> 
+  right_join(newest_plots, by = c("INVYR", "PLOT")) |> 
+  group_by(PLOT) |> 
+  summarise(
+    yr = mean(INVYR, na.rm = TRUE),
+    plot = mean(PLOT, na.rm = TRUE),
+    lat = mean(LAT, na.rm = TRUE),
+    long = mean(LON, na.rm = TRUE),
+    n_trees = n(),
+    agb = sum(DRYBIO_AG, na.rm = TRUE)
+  ) |> 
+  collect() |> 
+  mutate(
+    # Because of how we joined, `n_trees` is always at least 1 -- 
+    # even if there are 0 trees
+    n_trees = ifelse(is.na(agb) & n_trees == 1, 0, n_trees),
+    agb = ifelse(is.na(agb), 0, agb)
+  )
+
+DBI::dbDisconnect(con)
+
+readr::write_csv(plot_measurements, "plot_measurements.csv")
+```
+
+</details>
+
+With that pre-processing done, it's time for us to load our data and turn it into an sf object. We're going to reproject our data to use a coordinate reference system that the US government tends to use for national data products, like the FIA:
+
+```{r}
+#| message: false
+library(sf)
+
+invisible(sf_proj_network(TRUE))
+
+plot_measurements <- readr::read_csv("https://www.mm218.dev/junkdrawer/riemann/plot_measurements.csv") |> 
```

Comment:
Look at how we use `read_csv()` for the file in this directory at `/start/case-study`:
https://github.com/tidymodels/tidymodels.org/tree/main/content/start/case-study

That should work for this as well.